### PR TITLE
feature(zc): add EDAC and modify config of td3bc

### PR DESCRIPTION
**Pull request by Super1ce**
## Description
Add EDAC algorithm. Modify config of td3bc and cql.

## TODO
Add EDAC algorithm include policy, model and head.
Modify config of td3bc and cql including obs space , action space and dataset.

## Check List

- [ ] merge the latest version source branch/repo, and resolve all the conflicts
- [ ] pass style check
- [ ] pass all the tests

### CODE REVIEW

File: ding/model/template/edac.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+                action_shape,
+                actor_head_layer_num,
+                sigma_type='conditioned',
+                activation=activation,
+                norm_type=norm_type
+            )
+        )
+
+        critic_input_size = obs_shape + action_shape
+        self.critic = nn.Sequential(
```
Comment (Cloud-Pku) [inline ding/model/template/edac.py:71]: This should require a backbone network to encode the original obs.
And optionally, you can make actor and critic share one obs encoder like [this](https://github.com/opendilab/DI-engine/blob/a580019732bbfae6c628eddb16b00fa70fc16a1d/ding/model/template/qac.py#L67).

File: ding/policy/edac.py [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+        )
+        self._optimizer_policy = Adam(
+            self._model.actor.parameters(),
+            lr=self._cfg.learn.learning_rate_policy,
+        )
+
+        self._gamma = self._cfg.learn.discount_factor
+
+        init_w = self._cfg.learn.init_w
+        self._model.actor[2].mu.weight.data.uniform_(-init_w, init_w)
```
Comment (Cloud-Pku) [inline ding/policy/edac.py:169]: Why initialize the actor head this way

File: ding/policy/edac.py [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+
+            q_pred_grads = torch.autograd.grad(q_pred_tile.sum(),pre_acs,retain_graph=True,create_graph=True)[0]
+            q_pred_grads = q_pred_grads / (torch.norm(q_pred_grads,p=2,dim=2).unsqueeze(-1) + 1e-10)
+            # [Ensemble_num,batch_size,act_dim] -> [batch_size,Ensemble_num,act_dim]
+            q_pred_grads = q_pred_grads.transpose(0,1)
+
+            q_pred_grads = q_pred_grads @ q_pred_grads.permute(0,2,1)
+            masks = torch.eye(self._cfg.model.ensemble_num,device=obs.device).unsqueeze(dim=0).repeat(q_pred_grads.size(0),1,1) 
+            q_pred_grads = (1 - masks) * q_pred_grads
+            grad_loss = torch.mean(torch.sum(q_pred_grads,dim=(1,2))) / (self._cfg.model.ensemble_num - 1)
```
Comment (Cloud-Pku) [inline ding/policy/edac.py:275]: grad_loss = torch.mean(torch.sum(q_pred_grads,dim=(1,2))) * **self._eta** / (self._cfg.model.ensemble_num - 1)

**Review by Cloud-Pku — CHANGES_REQUESTED** [review]
